src/transformers/benchmark/benchmark.py

```diff
@@ -0,0 +1,122 @@
+# coding=utf-8
+# Copyright 2018 The HuggingFace Inc. team.
+# Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Benchmarking the library on inference and training in PyTorch """
+
+
+import inspect
+import timeit
+
+from transformers import MODEL_MAPPING, MODEL_WITH_LM_HEAD_MAPPING, PretrainedConfig, is_torch_available
+
+from .benchmark_utils import Benchmarks, Memory, start_memory_tracing, stop_memory_tracing
+
+
+if is_torch_available():
+    import torch
+    from .benchmark_args import PyTorchBenchmarkArguments
+
+
+class PyTorchBenchmarks(Benchmarks):
+
+    args: PyTorchBenchmarkArguments
+    configs: PretrainedConfig
+
+    def train(self, model_name, batch_size, sequence_length, trace_memory=False):
+        config = self.config_dict[model_name]
+        model = MODEL_WITH_LM_HEAD_MAPPING[config.__class__](config)
+        model.to(self.args.device)
+        model.train()
+
+        input_ids = torch.randint(
+            model.config.vocab_size, (batch_size, sequence_length), dtype=torch.long, device=self.args.device
+        )
+
+        def compute_loss_and_backprob():
+            # TODO: Not all models call labels argument labels => this hack using the function signature should be corrected once all models have a common name for labels
+            function_argument_names = inspect.getfullargspec(model.forward).args
```

Comment:
@julien-c - This is currently done because we have different names for `labels` (I think `Trainer` has the same issue).

Reply:
Yes, I would be in favor of that!

Reply:
would be great